### ISSUE TITLE
Support React v16

### ISSUE THREAD

**Issue by viztor** [JakeSidSmith/react-reorderable#94]
Now with React v16
it will throw exception createClass not a function.

**Comment by JakeSidSmith** [comment]
@viztor this lib uses the `create-react-class` module, which should still be compatible with React 16.

If not, I will be re-writing v3 in ES6 / TypeScript, and so will using actual classes.

**Comment by dargue3** [comment]
+1

I don't believe this works with React 16, unless you're saying we need to include `create-react-class` in our dependencies?

**Comment by JakeSidSmith** [comment]
It is a dep of react-reorder: https://github.com/JakeSidSmith/react-reorder/blob/master/package.json#L34

Can someone explain what exactly is the issue with React 16? :)

**Comment by AntCrick** [comment]
+1 Though I think it could be an issue with the version that is on npm as i used npm install to install this and it's installed `react-reorder@3.0.0-alpha.4` and the react-reorder package.json doesn't have create-react-class in the dependencies that i can see so could be that maybe?

**Comment by JakeSidSmith** [comment]
Ah, that's it! Thanks @AntCrick will update now and publish a new alpha version.

**Comment by JakeSidSmith** [comment]
@AntCrick @dargue3 @viztor have published a fix at `3.0.0-alpha.5`.

Please try this out, and if your issue is fixed, then close this issue. 🙂

**Comment by birdy90** [comment]
I get `Uncaught TypeError: Cannot read property 'oneOfType' of undefined` because of 
```JavaScript
   ...
   var PropTypes = React.PropTypes;
   Reorder.propTypes = {
      component: PropTypes.oneOfType([PropTypes.func, PropTypes.string]),
   ...
```
And as it's said here `https://reactjs.org/docs/typechecking-with-proptypes.html` it is deprecated and moved to some package. What should I do to handle this problem?

**Comment by JakeSidSmith** [comment]
Ah, poop. So many changes to make. Will try to get this sorted shortly.

**Comment by birdy90** [comment]
Maybe it will help: [there's](https://reactjs.org/blog/2017/04/07/react-v15.5.0.html#migrating-from-reactproptypes)  recommendations for migrating

**Comment by JakeSidSmith** [comment]
@birdy90 published some fixes at `3.0.0-alpha.6`.

**Comment by JakeSidSmith** [comment]
@AntCrick @dargue3 @viztor should now work nicely with React 16.
